### DISCUSSION TITLE
init.d start/stop script

### DISCUSSION THREAD

**Discussion by cnseubert** [inbucket/inbucket#523] [Q&A] [answered]
I don't want to use the docker image, and was able to get the inbucket to run fine, however I'm trying to get it to run from init.d on a system-v system . the provided .service script doesn't work for obvious reasons, and I've made an init.d service file with the appropriate environment settings in an /etc/default file.
However when I start the service using the init.d script, the inbucket process does not see the environment vars that I've set in the default file , nor when I set them directly in the init script.
Any suggestions would be much appreciated.

**Comment by jhillyerd** [comment, marked as answer]
Unlikely this is useful, but [this was the init script](https://github.com/inbucket/inbucket/blob/a246be6e5fe08b8695bf4eca516cca529201acb6/etc/redhat-el6/inbucket-init.sh) inbucket used way back when.  Unfortunately, it predates the switch from config file to env vars.

I haven't worked with init in years, so I can't really give specific advice.  Are you exporting the vars so that Inbucket can see them, and using the INBUCKET_ naming scheme?  example start script I use for dev https://github.com/inbucket/inbucket/blob/main/etc/dev-start.sh

**Comment by cnseubert** [comment]
ok so it turns out I don't know why but the first go I had at the script, exporting the env vars didn't work, and after much flailing about, and being told to try that again.. now it works! I just export the env vars in the init script and now it's happy.
Thanks for the assist!
